docs: March release notes (#9732)
adobe/react-spectrum: packages/dev/s2-docs/pages/react-aria/releases/v1-16-0.mdx

* Support collections nested within Activities (React 19.2) - [@mhornbacher](https://github.com/mhornbacher) - [PR](https://github.com/adobe/react-spectrum/pull/9300)
* Export LongPressEvent and MoveEvent types - [@poetryofcode](https://github.com/poetryofcode) - [PR](https://github.com/adobe/react-spectrum/pull/9522)
* Only handle blur event for resizing viewport on iOS for proper viewport dimensions - [@lixiaoyan](https://github.com/lixiaoyan) - [PR](https://github.com/adobe/react-spectrum/pull/9393)
* Use KeyboardProps in Keyboard component and context - [@artola](https://github.com/artola) - [PR](https://github.com/adobe/react-spectrum/pull/9668)
* Prevent docs crash by always placing `template` element children into the `template`'s content - [@LFDanLu](https://github.com/LFDanLu) - [PR](https://github.com/adobe/react-spectrum/pull/9385), [PR](https://github.com/adobe/react-spectrum/pull/9703), [PR](https://github.com/adobe/react-spectrum/pull/9712)
### Button
* Allow tooltip to show on hover when Button has isPending (#9460) - [@albertdugba](https://github.com/albertdugba) - [PR](https://github.com/adobe/react-spectrum/pull/9619)
### Calendar

> * Prevent setFocusedDate from stealing focus from elements outside of Calendar - [@AryanBagade](https://github.com/AryanBagade) - [PR](https://github.com/adobe/react-spectrum/pull/9463)

Reviewer: ```suggestion
* Prevent `setFocusedDate` from stealing focus from elements outside of Calendar - [@AryanBagade](https://github.com/AryanBagade) - [PR](https://github.com/adobe/react-spectrum/pull/9463)
```

### Tabs
* Prevent setting `selectedKey=null` to avoid crash in Tabs - [@abeljohn](https://github.com/abeljohn) - [PR](https://github.com/adobe/react-spectrum/pull/9585)
### Table
* Fix type so TableHeaderProps.columns properly accepts readonly arrays - [@sonsu-lee](https://github.com/sonsu-lee) - [PR](https://github.com/adobe/react-spectrum/pull/9611)
### Tree
* Centralize expandedKeys logic in TreeCollection - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/9711)
* Tree keyboard drag and drop should skip content nodes - [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/9724)
### Tooltip
* Respect delay when open() is called twice in useTooltipTriggerState - [@gohsue](https://github.com/gohsue) - [PR](https://github.com/adobe/react-spectrum/pull/9518)
### Tree
* Add support for tree sections - [@yihuiliao](https://github.com/yihuiliao) - [PR](https://github.com/adobe/react-spectrum/pull/9013)

## Under Construction

> * Shadow DOM

Reviewer 2: ```suggestion
* Shadow DOM - behind the feature flag `enableShadowDOM`
```

>   * Improve performance of node.contain calls - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/9536)

Reviewer: ```suggestion
  * Improve performance of `node.contain` calls - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/9536)
```

>   * Update all active element queries for shadow dom support - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/9608)

Reviewer: ```suggestion
  * Update all active element queries for shadow DOM support - [@snowystinger](https://github.com/snowystinger) - [PR](https://github.com/adobe/react-spectrum/pull/9608)
```

the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
OF ANY KIND, either express or implied. See the License for the specific language
governing permissions and limitations under the License. */}

import {Layout} from '../../../src/Layout';
export default Layout;

import docs from 'docs:@react-spectrum/s2';

export const hideNav = true;
export const section = 'Releases';
export const tags = ['release', 'React Aria'];
export const date = 'March 3, 2026';
export const title = 'v1.16.0';
export const description = 'In today’s release we are excited to announce support for multi-select ComboBox, section support in Tree, and a variety of improvements to overlay positioning, scroll into view behavior, and more!'
export const isSubpage = true;

# v1.16.0

In today’s release, we are excited to announce support for [multi-select ComboBox](../ComboBox), one of our most requested features! You can now use selectionMode="multiple"
on ComboBox to select multiple options and customize how to display the selected items via ComboBoxValue, e.g. using a TagGroup. We’ve also added [section support](../Tree#sections-)

> in Tree: simply use TreeSection and TreeHeader to group tree items into labeled sections. This release also includes improvements to overlay positioning, scroll into view behavior,

Reviewer 3: ```suggestion
in Tree: simply use `TreeSection` and `TreeHeader` to group tree items into labeled sections. This release also includes improvements to overlay positioning, scroll into view behavior,
```

and fixes to address various crashes.

> As always, a special thank you to all of our contributors for your feedback, features, and fixes!

Reviewer 4: ```suggestion
This release contains one of the highest number of contributions to date. A big thank you to our contributors for all your feedback, features and fixes!
```

* Fix `scrollIntoView` so it accounts for scrollbar width & RTL layout - [@nwidynski](https://github.com/nwidynski) - [PR](https://github.com/adobe/react-spectrum/pull/9634)
### Tabs
* Prevent setting `selectedKey=null` to avoid crash in Tabs - [@abeljohn](https://github.com/abeljohn) - [PR](https://github.com/adobe/react-spectrum/pull/9585)
### Table
* Fix type so TableHeaderProps.columns properly accepts readonly arrays - [@sonsu-lee](https://github.com/sonsu-lee) - [PR](https://github.com/adobe/react-spectrum/pull/9611)
### Tooltip
* Respect delay when open() is called twice in useTooltipTriggerState - [@gohsue](https://github.com/gohsue) - [PR](https://github.com/adobe/react-spectrum/pull/9518)
### Tree
* Add support for tree sections - [@yihuiliao](https://github.com/yihuiliao) - [PR](https://github.com/adobe/react-spectrum/pull/9013), [@devongovett](https://github.com/devongovett) - [PR](https://github.com/adobe/react-spectrum/pull/9711), [PR](https://github.com/adobe/react-spectrum/pull/9724)

> ## Under Construction

Reviewer 3: I meant shadow dom

> ## Under Construction

Reviewer 5: Oh MAN!

I know you are not advertising. But I have been following this work like some crazed stalker. And I know this isn't perfect, and it's a WIP, but it's enough that it makes my life incredibly less complicated. Somebody, please, hug @snowystinger for me!!!
